src/functions/functionDefinitions.ts

```diff
@@ -73,6 +73,52 @@ const functionDefinitions = (functionsDir: string): FunctionDefinitions => {
   );
 };
 
+const stringifyDefinitions = (definitions: FunctionDefinitions): string => {
+  const updatedDefinitions = Object.keys(definitions).reduce((acc, name) => {
+    const definition = definitions[name];
+    return {
+      ...acc,
+      [name]: {
+        ...definition,
+        options: JSON.stringify(definition.options),
+      },
+    };
+  }, {});
+
+  return JSON.stringify(updatedDefinitions);
+};
+
+/* @doc newFunctionDefinition
+  Creates a new folder for the given name and fills that folder with a function.json file
+  and an index.js.
+*/
+const newFunctionDefinition = (
+  functionsDir: string,
+  functionName: string,
+): void => {
+  const functionDefName = functionName.replace(/-./g, x => x.toUpperCase()[1]);
+  const functionDir = path.join(functionsDir, functionName);
+  fs.mkdirpSync(functionDir);
```

Comment:
since IO can be very unreliable, this is usually a good place to try catch some errors so that when things blow up you have some context (i.e what file we were trying to read)